.github/workflows/ci.yml

```
@@ -1,17 +1,73 @@
 name: wsrpc
-on: push
+
+on: 
+  push:
+    
 jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version-file: 'go.mod'
+      
+        ##
+        # XXX: change this to the official action once multiple --out-format args are supported.
+        # See: https://github.com/golangci/golangci-lint-action/issues/612
+        ##
+      - name: golangci-lint
+        uses: smartcontractkit/golangci-lint-action@54ab6c5f11d66a92d14c3f7cc41ea13f676644bd # feature/multiple-output-formats-backup
+        with:
+          # Optional: version of golangci-lint to use in form of v1.2 or v1.2.3 or `latest` to use the latest version
+          version: v1.52.1
+          
+          # Optional: working directory, useful for monorepos
+          # working-directory: 
+
+          # Optional: golangci-lint command line arguments.
+          allow-extra-out-format-args: true
+          args: --timeout=2m0s --out-format checkstyle:golangci-lint-report.xml
+
+          # Optional: show only new issues if it's a pull request. The default value is `false`.
+          only-new-issues: true
+
+          # Optional: if set to true then the action will use pre-installed Go.
+          # skip-go-installation: true
+
+          # Optional: if set to true then the action don't cache or restore ~/go/pkg.
+          # skip-pkg-cache: true
+
+          # Optional: if set to true then the action don't cache or restore ~/.cache/go-build.
+          # skip-build-cache: true
+
+      - name: Verify lint report artifact
```

Why 'Verify'? This is about printing the results for humans, if the file exists.
```suggestion
      - name: Print lint report artifact
```